doc/SONiC_OC_NTP_HLD.md

```diff
@@ -0,0 +1,952 @@
+c.if "default" is again configured as NTP vrf, ntp service starts in default vrf context      
+     
+##### When mgmt VRF is not configured     
+ntp service always starts in default vrf context      
+      
+##### NTP source interface related      
+a.if a NTP source interface has IP address configured, the IP address will be used as source ip for all NTP packets exchanged with the respective NTP servers/clients    
+b.if a NTP source interface has no IP address configured, it is not being considered as an NTP source interface     
+ 
+#### 1.1.1.6 NTP authentication configuration
+NTP authentication enables an NTP client or peer to authenticate time received from their servers and peers.
+
+##### ntp authenticate
+```
+ntp authenticate
+```
+
+NTP server and NTP client use this command to enable the NTP authentication feature.
+        
+##### ntp authentication-key
+```
+ntp authentication-key 1 md5 "ntp client 1"
+
+ntp authentication-key 2 md5 ntp_client2
+```
+    
+NTP client uses this command to define an authentication key with key number, authentication type and password.     
+The key number is from 1 to 65535.     
+The authentication type supported is MD5, SHA1 and SHA2-256.     
+The password is configured with plaintext the first time. In runnning-configuration, it is encrypted and is indicated with "encrypted" at the end. Authentication key can then be configured with the encrypted format and "encrypted" flag. 
+      
+    
+##### ntp trusted key
+```
+ntp trusted-key 1
+
+ntp trusted-key 2
+```
+    
+NTP server uses this command to add a list of key numbers that the NTP server must provide in its NTP packets in order for the NTP clients to synchronize to it.
+NTP client must configure a list with this command if it desires to authenticate the NTP server with any of the key number.      
+
+##### ntp server key
+```
+ntp server 99.1.1.1 key 1
+```
+     
+NTP client uses this command to configure the key expected from a specific NTP server.        
+
+### 1.1.2 Backend mechanisms to support configuration and get
+
+#### 1.1.2.1 add/delete NTP server
+This creates or deletes a NTP server entry in the Redis ConfigDB.
+
+```
+  "NTP_SERVER|10.11.0.1": {
+    "type": "hash",
+    "value": {
+      "NULL": "NULL"
+    }
+  },
+  "NTP_SERVER|2001:aa:aa::a": {
+    "type": "hash",
+    "value": {
+      "NULL": "NULL"
+    }
+  },
+  "NTP_SERVER|pool.ntp.org": {
+    "type": "hash",
+    "value": {
+      "NULL": "NULL"
+    }
+  }
+```
+
+A change in the NTP_SERVER entry triggers hostcfgd to start the NTP configuration script, which in turn writes each NTP server to the ntp.conf and then restart the ntp service.   
+               
+
+#### 1.1.2.2 add/delete NTP source    
+
+This creates or deletes a global NTP source interface entry in the Redis ConfigDB.      
+
+```
+  "NTP|global": {
+    "type": "hash",
+    "value": {
+      "src_intf": "Ethernet36"

```

Comment:
Per 0.5 version src_intf would be a list now (with the trailing  \@ symbol) ?  Eg: "\src_intf@": "Loopback0,eth0"